feat: Add Farcaster miniapp support for identity verification flow #10

vortex-hue opened the pull request:
# Description

This PR implements comprehensive Farcaster miniapp support for the identity verification flow, enabling seamless face verification within Farcaster miniapps with proper navigation and universal link handling. The implementation includes automatic miniapp detection, smart navigation, and robust response handling while fixing critical build system issues that were preventing successful compilation.

The changes address the need for better Farcaster miniapp integration by implementing the official `@farcaster/miniapp-sdk`, adding universal link support for mobile/native compatibility, and creating utilities for handling post-verification flow continuation. Additionally, this PR resolves significant build configuration issues including Node.js dependency conflicts in browser builds and security vulnerabilities related to environment variable exposure.

**Dependencies added:**
- `@farcaster/miniapp-sdk` (peer dependency)
- Browser polyfills for Node.js modules (crypto-browserify, stream-browserify, buffer)

**About** #10

### Browser Demo Video: [Loom GoodSDK Forecaster Support](https://www.loom.com/share/4c4b5f71d7c74709a1081336e4a96a93?sid=f9e87733-4945-4049-955f-2b7611bfbca1)

### Mobile Demo Video: [Loom GoodSDK Forecaster Support](https://files.fm/f/zx49en7rgs)

I would like to know how I can demo on a mobile please, but I was able to test the demo on a browser.

# Checklist:

**Key Implementation Details:**

**Farcaster Integration:**

- [x] `isInFarcasterMiniApp()` - Async detection using official SDK
- [x] `navigateToFaceVerification()` - Smart navigation with automatic environment detection
- [x] `openUrlInFarcaster()` - Official SDK integration with `openUrl` method
- [ ] `handleVerificationResponse()` - Generic utility for post-verification flow
- [x] `createUniversalLinkCallback()` - Universal link support for mobile/native apps

**Build System Fixes:**
- Externalized Node.js modules (`crypto`, `http`, `https`, `stream`, `zlib`, etc.) for browser compatibility
- Fixed security vulnerabilities by replacing full `process.env` exposure with specific variables
- Optimized tsup configuration for ESM-only builds to avoid IIFE conflicts
- Restructured package dependencies to use peer dependencies for heavy libraries

**Sensitive Files Modified:**

These are some of the files I modified after my implementation due to build error I was getting, because it was building for browser, meanwhile some Node.js functions are also present, I'd appreciate if there's another way I could resolve the error without modifying these files to avoid a merge conflict.

- Core SDK: `packages/citizen-sdk/src/index.ts`, `packages/citizen-sdk/src/utils/auth.ts`
- Build configs: `packages/ui-components/tsup.config.claim.ts`, `packages/ui-components/package.json`
- Demo apps: `apps/demo-identity-app/vite.config.mts`, `apps/demo-webcomponents/vite.config.mts`
- Components: `apps/demo-identity-app/src/components/VerifyButton.tsx`
- Types: `apps/demo-identity-app/src/globals.d.ts`
- yarn.lock - Updated dependencies

vortex-hue committed:
Feat: Installed farcaster package

vortex-hue committed:
fix: build errors

vortex-hue committed:
feat: Add Farcaster miniapp support for identity flow with build fixes

vortex-hue committed:
rm eslintrc

vortex-hue committed:
refactor: Address code review feedback with improved architecture across utilities

L03TJ3 reviewed (changes requested):
I also think I miss seeing handling of the 'verified' param that you get when returning from the face-verification?

vortex-hue committed:
fix: Address code review complaints

vortex-hue commented on packages/build-config/index.ts:
```diff
@@ -0,0 +1,41 @@
+import { builtinModules } from "module"
```
I added it for experimental shared config purpose, it isn't being used anywhere

vortex-hue replied:
hi lewis, I think I'd bring this file back to manage the shared configs considering it's a mono repo, and then import it across all places it's needed, I've already done that.

vortex-hue commented on apps/demo-webcomponents/vite.config.mts:
```diff
@@ -1,6 +1,7 @@
 import { defineConfig } from "vite"
```
fixed

vortex-hue commented on packages/citizen-sdk/src/sdks/viem-identity-sdk.ts:
```diff
@@ -226,7 +228,15 @@ export class IdentitySDK {
       if (callbackUrl) {
-        params[popupMode ? "cbu" : "rdu"] = callbackUrl
+        // Import utility functions for universal link support
+        const { createUniversalLinkCallback } = await import("../utils/auth")
```
done

vortex-hue commented on packages/citizen-sdk/src/sdks/viem-identity-sdk.ts:
```diff
@@ -263,4 +273,23 @@ export class IdentitySDK {
+    chainId?: number,
+    forceFarcasterNavigation?: boolean,
+  ): Promise<void> {
+    const { farcasterNavigation } = await import("../utils/farcaster-navigation")
```
done too

sirpy commented on apps/demo-identity-app/vite.config.mts:
```diff
@@ -29,6 +43,15 @@ export default defineConfig({
   define: {
     "process.browser": true,
-    "process.env": process.env,
+    // Only define specific env vars instead of the entire process.env object
+    "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV || "development"),
```
why? restore

sirpy commented on apps/demo-identity-app/vite.config.mts:
```diff
@@ -29,6 +43,15 @@ export default defineConfig({
+    // Only define specific env vars instead of the entire process.env object
+    "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV || "development"),
   },
+  build: {
+    rollupOptions: {
+      external: nodeBuiltins,
+      output: {
+        globals: nodeBuiltinGlobals
+      }
+    }
+  }
```
why?

sirpy commented on apps/demo-webcomponents/vite.config.mts:
```diff
@@ -23,6 +25,12 @@ export default defineConfig({
+    // Only define specific env vars instead of the entire process.env object
+    "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV || "development"),
   },
+  build: {
+    rollupOptions: {
+      external: ["@goodsdks/citizen-sdk"]
+    }
+  }
```
doesnt make sense this has to be included

sirpy commented on packages/citizen-sdk/src/sdks/viem-claim-sdk.ts:
```diff
@@ -266,18 +266,12 @@ export class ClaimSDK {
-      )
-    }
+    // Use the new navigation method that handles Farcaster automatically
+    await this.identitySDK.navigateToFaceVerification(false, this.rdu, 42220)
```
the content of navigatetoface should be here... it doesnt make sense to have a one line method calling another method

L03TJ3 replied:
didn't we agree that we wanted to prevent having to maintain same code in different places? @sirpy

L03TJ3 replied:
@vortex-hue but this could be included in the regular claim-flow, does not have to be a separate method

sirpy commented on packages/citizen-sdk/src/sdks/viem-identity-sdk.ts:
```diff
@@ -263,4 +272,21 @@ export class IdentitySDK {
+    forceFarcasterNavigation?: boolean,
+  ): Promise<void> {
+    const fvLink = await this.generateFVLink(popupMode, callbackUrl, chainId)
+    await farcasterNavigation.navigateToFaceVerification(fvLink, popupMode, callbackUrl, forceFarcasterNavigation)
```
inline content again

sirpy commented on packages/citizen-sdk/src/utils/auth.ts:
```diff
@@ -1,6 +1,156 @@
+  try {
+    // Basic detection without the async SDK call
+    return (
+      window.location.href.includes("farcaster") ||
+      window.location.href.includes("miniapp") ||
+      // Check if we're in an iframe which is common for miniapps
+      window.self !== window.top
```
that can return many false positives, not sure why this is needed

L03TJ3 replied:
@vortex-hue

sirpy commented on packages/citizen-sdk/src/utils/auth.ts:
```diff
@@ -1,6 +1,156 @@
+ * @param fallbackToNewTab - Whether to open in new tab as fallback (default: true)
+ */
+export async function openUrlInFarcaster(url: string, fallbackToNewTab: boolean = true): Promise<void> {
+  if (typeof window === "undefined") {
+    throw new Error("URL opening is only supported in browser environments.");
+  }
```
duplicate you already check if in farcaster app

L03TJ3 replied:
@vortex-hue

sirpy commented on packages/citizen-sdk/src/utils/auth.ts:
```diff
@@ -1,6 +1,156 @@
+ * @param additionalParams - Additional parameters to include
+ * @returns A universal link compatible URL
+ */
+export function createUniversalLinkCallback(
```
i dont understand the logic or why this is required. the app owner passes the required callback url as is in its full required form

L03TJ3 replied:
I requested this flow so that the verified param can be appened and a app can handle the redirectBack from face-verification.
You think this should be handled different?

L03TJ3 replied:
@vortex-hue pending on @sirpy response, make sure the verified param is handled.
example here: https://github.com/GoodDollar/GoodWeb3-Mono/blob/70557ec8977959199e7a95e4872b6d3bca3e8327/packages/good-design/src/core/buttons/ClaimButton.tsx#L31

sirpy commented on packages/citizen-sdk/src/utils/farcaster-navigation.ts:
```diff
@@ -0,0 +1,103 @@
+  /**
+   * Create a universal link compatible callback URL
+   */
+  private createUniversalLinkCallback(
```
duplicate code

sirpy commented on packages/citizen-sdk/src/utils/farcaster-navigation.ts:
```diff
@@ -0,0 +1,103 @@
+    url: string, 
+    fallbackToNewTab: boolean = true
+  ): Promise<void> {
+    if (typeof window === "undefined") {
```
duplicate code. if in farcaster then it is obvious theres a window this check is redundant

sirpy commented on packages/ui-components/tsup.config.claim.ts:
```diff
@@ -1,8 +1,14 @@
 export default defineConfig({
   entry: ["src/index.ts"],
-  format: ["esm", "iife"],
+  format: ["esm"], // Only build ESM format since IIFE has issues with external dependencies
```
iife is required for web components

sirpy replied:
It worked perfectly fine before your changes
@L03TJ3  please chime in

L03TJ3 replied:
1. Once something like this from your pov breaks the suggested approach is to reach out and ask if we maybe know about a bug or not. The answer would have been: this worked fine before, your changes should not have affected the behavior of the UI components.

2. you chose to use dynamic import of farcaster-sdk? why, I don't see any need for this caching/lazy-loading of the sdk

3. its the dynamic import that is causing issues building the ui-component. adding @farcaster/miniapp-sdk as dependency to citizen-sdk, removing the dynamic/lazy-load import and replace with module import, will have both citizen-sdk and ui-components build without any changes to configuration necessary

sirpy commented on packages/ui-components/tsup.config.claim.ts:
```diff
@@ -12,4 +18,20 @@ export default defineConfig({
+    ...nodeBuiltins
+  ],
+  // Only bundle UI-specific code, not blockchain/crypto libraries
+  noExternal: [
```
this are for sure do not need to be included. they are peer deps

sirpy commented on packages/ui-components/tsup.config.claim.ts:
```diff
@@ -12,4 +18,20 @@ export default defineConfig({
   target: "ESNext",
   outDir: "dist",
+  // Mark heavy dependencies as external to avoid bundling Node.js modules
+  external: [
```
why is this suddenly required?

sirpy reviewed (changes requested):
1. The code needs to be organized better. Many duplicates, single line functions etc.
2. since farcaster always opens a new tab/window it is possibly best to force cbu, ie popup mode, where the tab/window will be closed after FV is done without redirect back.
3. for cbu need to update the check is verified, to not only check the url for isVerified param (since there is no redirect with that param in cbu mode) but also to accept as input from developer the wallet address and check on-chain if wallet is now whitelisted

sirpy commented:
@L03TJ3 the non custodial is just a duplicate code requiring now multiple maintenance. it should be integrated into the regular sdk

vortex-hue committed:
fix: working on resolving the reviews made

vortex-hue committed:
fix: removed dynamic import, and reverted some configs to the default and added farcaster/miniapp-sdk as dependency to citizen-sdk

sirpy commented:
> Hi @L03TJ3 @sirpy, I've implemented all the corrections, but am yet to still understand how to test it in mini app, should I turn the example test app into a forecaster mini app?

either it should work in both (web/farcaster) or you can create a similar farcaster miniapp

vortex-hue committed:
pulled update

vortex-hue commented:
Hi @sirpy @L03TJ3 , I've been able to test it in web/farcaster environments, here's the loom: [Demo in Forecaster & Web](https://www.loom.com/share/4c4b5f71d7c74709a1081336e4a96a93?sid=f9e87733-4945-4049-955f-2b7611bfbca1), also thanks Lewis for other time, and Emiri helped me alot too

vortex-hue commented:
Do let me know if there's anything else pending, thanks alot.

sirpy commented:
@vortex-hue Please make sure you went over all the open items and made the relevant changes.
The loom looks nice! can you also test it on the mobile farcaster app?

@L03TJ3 should review this by end of week

L03TJ3 commented on apps/demo-identity-app/vite.config.mts:
```diff
@@ -13,6 +13,8 @@ if (process.env.HTTPS === "true") {
   https = false;
 }
 
+
```
not needed changes

vortex-hue committed:
fix: Addressed all outstanding code review feedback.

vortex-hue committed:
fix: resolved all code changes requested

vortex-hue committed:
feat: implement shared build config utility for Node.js built-ins and also created packages/build-config/index.ts with dynamic Node.js built-ins generation

L03TJ3 reviewed (changes requested):
On your last commit:
PR's should be scoped towards what you are trying to solve.
no additional work is needed. if you want those type of contributions you can put forward a new PR for it.

None of these changes seem needed as on master branch, all the apps and packages deploy and build as expected.

If you encounter build errors or deployment errors, please discuss what you are facing.
Revert these please

vortex-hue committed:
revert: removed the shared build config changes and other changes

vortex-hue committed:
fc:frame meta tag

vortex-hue committed:
fix: refined the codebase

vortex-hue commented:
Hi @sirpy @L03TJ3 , I've been able to test it on mobile using the demo-app in the mono repo and here's the link to the loom: [moile testing](https://www.loom.com/share/eabea60d41bd4218b6198dd4bac7af38), i noticed the vercel face verification system doesn't redirect back throughout my testing, and from abit of research I did, I found out it was because our demo-app isn't recognised by farcaster since we're just tunnelling it under the preview section.

vortex-hue committed:
fix:patching up the codebase

vortex-hue committed:
Revert "fix:patching up the codebase"

This reverts commit 8616d89cbad1adee3dd9ec68488962969f2c51a4.

vortex-hue committed:
revert: reverted all changes requested

vortex-hue committed:
revert: reverted all changes requested

vortex-hue committed:
revert: reverted all changes requested

vortex-hue committed:
fix: revert changes requested

vortex-hue committed:
fix: revert changes requested

vortex-hue committed:
fix: revert changes affecting mobile

vortex-hue committed:
feat: implemented the universal link using farcaster docs

vortex-hue committed:
fix: resolving previous issues

vortex-hue committed:
fix: resolved redirection concerns and fixed bugs

vortex-hue committed:
Merge remote-tracking branch 'origin/main' into add-farcaster-support-for-the-identity-flow

L03TJ3 reviewed (changes requested):
the project does not build:

│ src/sdks/viem-claim-sdk.ts:20:2: ERROR: No matching export in "src/constants.ts" for import "contractAddresses"
│ src/sdks/viem-claim-sdk.ts:22:26: ERROR: No matching export in "src/constants.ts" for import "getGasPrice"
│ src/utils/auth.ts:2:50: ERROR: No matching export in "src/constants.ts" for import "contractAddresses"

vortex-hue committed:
fix: build id working well now

vortex-hue committed:
fix: remove the unused interface of VerifyButtonProps

vortex-hue commented on apps/demo-identity-app/src/App.tsx:
```diff
@@ -40,14 +43,166 @@ const App: React.FC = () => {
   const { sdk: identitySDK } = useIdentitySDK("development")
 
   useEffect(() => {
```
do you mean I should revert the file to how it was initially? and remove every code I added relating to the forecaster support?

vortex-hue committed:
fix: removed the used import { createUniversalLinkCallback } from ../utils/auth

vortex-hue committed:
fix: revert apps/demo-identity-app/src/App.tsx and verifybutton.tsx to original state

vortex-hue committed:
Merge upstream/main into add-farcaster-support-for-the-identity-flow

Resolved conflicts in:
- packages/citizen-sdk/src/sdks/viem-claim-sdk.ts
- packages/citizen-sdk/src/sdks/viem-identity-sdk.ts

Merged changes:
- Kept Farcaster miniapp support (Universal Links, callback URLs)
- Integrated new chain management system with fallbacks
- Added RPC fallback mechanism
- Refactored faucet trigger utility
- Updated to use identitySDK.navigateToFaceVerification() instead of fvRedirect()

vortex-hue committed:
fix: update auth.ts to use chainConfigs instead of contractAddresses

After merging upstream/main, the contract address structure changed from
contractAddresses to chainConfigs. Updated isAddressWhitelisted function
to use the new chainConfigs structure.

vortex-hue committed:
Merge remote-tracking branch 'upstream/main' into add-farcaster-support-for-the-identity-flow

vortex-hue committed:
refactor: apply engineering principles to Farcaster implementation

- Remove excessive console.log statements from auth utilities and SDKs
- Simplify error handling by removing redundant try-catch blocks
- Remove verbose JSDoc comments that state the obvious
- Consolidate duplicate navigation functions
- Improve code readability and maintainability
- Follow KISS, DRY, and clarity over cleverness principles

vortex-hue committed:
refactor: clean up demo app following engineering principles

- Remove unnecessary console.log/console.error statements
- Remove unused onVerificationSuccess prop from VerifyButton
- Simplify error handling in demo components
- Rename callBackExample to txConfirm for clarity
- Follow DRY and KISS principles

L03TJ3 commented on packages/citizen-sdk/src/constants.ts:
```diff
@@ -27,6 +27,25 @@ export const Envs: Record<string, Record<string, string>> = {
   },
 }
 
+/**
```
If these are just placeholders and not working this should be clarified in a readme and should be configurable options for developers to pass down
(we are not hosting a farcaster app, developers integrating the identitysdk/claim-sdk might)

L03TJ3 commented on packages/citizen-sdk/src/utils/auth.ts:
```diff
@@ -1,5 +1,253 @@
+  return universalLink;
+}
+
+export async function createVerificationCallbackUrl(
```
What is this supposed to be handling?

L03TJ3 commented on packages/citizen-sdk/src/sdks/viem-identity-sdk.ts:
```diff
@@ -106,168 +115,134 @@ export class IdentitySDK {
-   * @param authPeriod - The authentication period.
-   * @returns The identity expiry data.
-   */
   calculateIdentityExpiry(
```
Why got the comments removed?

L03TJ3 commented on packages/citizen-sdk/src/sdks/viem-identity-sdk.ts:
```diff
@@ -106,168 +115,134 @@ export class IdentitySDK {
+        );
+        params[popupMode ? "cbu" : "rdu"] = universalLinkCallback;
+      } else {
+        const callbackUrlWithParams = await createVerificationCallbackUrl(callbackUrl, {
```
What are these params used for, and why hardcode 'verified': true?

L03TJ3 reviewed (changes requested):
Summary of the comments:
A proposed solution in a PR to resolve an issue should tackle only what is relevant to your solution and should not introduced extra refactors or changes without an explicit reason

vortex-hue committed:
fix: address code review feedback

- Restore original fvRedirect behavior in ClaimSDK (window.location.href)
- Use fvDefaultChain instead of hardcoded chainId
- Restore error handling in submitAndWait
- Restore removed JSDoc comments
- Remove unused handleVerificationResponse functions
- Fix VerifyButton error handling
- Add FarcasterAppConfigs documentation to README
- Add JSDoc to createVerificationCallbackUrl

vortex-hue committed:
fix: address round 3 code review comments (restore comments, fix fvRedirect, restore error handling)

L03TJ3 commented on packages/citizen-sdk/src/sdks/viem-identity-sdk.ts:
```diff
@@ -106,168 +115,134 @@ export class IdentitySDK {
-      console.error("submitAndWait Error:", error)
-      throw new Error(`Failed to submit transaction: ${error.message}`)
-    }
+    return waitForTransactionReceipt(this.publicClient, { hash })
```
@vortex-hue comment not addresses

L03TJ3 commented on apps/demo-identity-app/src/App.tsx:
```diff
@@ -70,8 +70,8 @@ const App: React.FC = () => {
           setIsVerified(isWhitelisted ?? false)
-        } catch (error) {
-          console.error("Error checking whitelist:", error)
+        } catch {
```
maybe the setIsWhitelisted should have been added in the catch but there is no reason to remove the error log

L03TJ3 commented on packages/citizen-sdk/src/utils/auth.ts:
```diff
@@ -1,5 +1,176 @@
+  if (additionalParams) {
+    Object.entries(additionalParams).forEach(([key, value]) => {
+      url.searchParams.set(key, value);
+    });
```
I think the original request has not been understood.
we are not going to enforce users to create a particular endpoint.
What we provide is helpers to the `verified` param, for them to handle UX/UI.

you get a queryParam 'verified' (true/false).
this shows if someone face-verification has been succesfull.
all we need is a helper that parses it.
in the original issue a reference has been shared where we do this in our old sdk: https://github.com/GoodDollar/GoodWeb3-Mono/blob/9846f16b16f2d8406407dd48231ea7c9e33e751e/packages/good-design/src/core/buttons/ClaimButton.tsx#L31

vortex-hue committed:
fix: address PR review feedback from L03TJ3

- Restore try-catch in getWhitelistedRoot with meaningful error message
- Re-throw original error in submitAndWait to preserve error type
- Simplify createVerificationCallbackUrl (remove /verify path enforcement)
- Add parseVerificationResult helper for verified query param parsing
- Restore console.error in App.tsx catch block
- Remove unused navigateToFaceVerification method
- Remove commit-style comment in fvRedirect

sirpy commented on packages/citizen-sdk/src/sdks/viem-identity-sdk.ts:
```diff
@@ -158,108 +171,101 @@ export class IdentitySDK {
+      const farcasterAppConfig = this.farcasterConfig ?? FarcasterAppConfigs[this.env];
+
+      if (isInFarcaster && farcasterAppConfig) {
+        const farcasterConfig = farcasterAppConfig;
```
why are you reassigning?

vortex-hue committed:
fix: address round 2 PR review feedback from sirpy

- Restore all original JSDoc, try-catch, and error handling patterns
- Add resolveCallbackUrl to IdentitySDK for Farcaster/URL consolidation
- Remove duplicate Farcaster logic from ClaimSDK (delegate to IdentitySDK)
- Remove FarcasterAppConfigs (apps don't have envs in Farcaster)
- Remove redundant isAddressWhitelisted from auth utils
- Move @farcaster/miniapp-sdk to peerDependencies
- Restore VerifyButton original props and chain ID

vortex-hue committed:
chore: merge upstream/main and regenerate yarn.lock

vortex-hue committed:
feat: add navigateToFaceVerification with Farcaster-aware redirect

- Add navigateToFaceVerification() to IdentitySDK for Farcaster-aware FV navigation
- ClaimSDK.fvRedirect() delegates to IdentitySDK instead of using window.location.href
- Refactor FarcasterAppConfig: domain is now required, appId/appSlug optional
- Support domain-based Farcaster deep links for redirect-back flow
- Pass farcasterConfig through useIdentitySDK and useClaimSDK hooks
- VerifyButton uses navigateToFaceVerification() for correct navigation

sirpy reviewed (dismissed):
1. you should be able to use identity sdk with dev env, it checks liveness but shouldnt check for duplicates. 
2. in order to show it is working you need to create a demo mini app and then access this app via farcaster web and also via farcaster mobile. you can follow their developer guides how to test apps.

vortex-hue committed:
refactor: remove Farcaster config coupling from SDK internals

- Removed `farcasterConfig` from `IdentitySDK` options and class state
- Removed `farcasterConfig` param from `useIdentitySDK` and `useClaimSDK` hooks
- Reverted `generateFVLink` back to using `callbackUrl` directly without internal interception
- Removed `resolveCallbackUrl` helper method
- Reverted `ClaimSDK.fvRedirect()` to use `window.location.href`
- Added explicit `generateFarcasterCallback` method to `IdentitySDK`
- Updated `VerifyButton` to handle Farcaster context explicitly and generate callback URL locally

L03TJ3 commented:
@vortex-hue Hey, its been pending to include demo-video. (as requested on TG)
there are also now merge conflicts to resolve

vortex-hue committed:
Merge branch 'main' into add-farcaster-support-for-the-identity-flow

vortex-hue commented:
Hi Lewis, i've resolved the merge conflict (accepted incoming changes), and
also the demo videos for the mobile and browser is in the description of
the pr, do let me know if there's any other changes to be made to get the
pr merged?

On Tue, Mar 31, 2026 at 12:17 PM Lewis B ***@***.***> wrote:

> *L03TJ3* left a comment (GoodDollar/GoodSDKs#14)
> <https://github.com/GoodDollar/GoodSDKs/pull/14?email_source=notifications&email_token=AXYCBIBFB2M6JR45QLI5E2D4TOSLTA5CNFSNUABFM5UWIORPF5TWS5BNNB2WEL2JONZXKZKDN5WW2ZLOOQXTIMJWGE4DQNRQGU42M4TFMFZW63VHNVSW45DJN5XKKZLWMVXHJNLQOJPWG33NNVSW45C7N5YGK3S7MNWGSY3L#issuecomment-4161886059>
>
> @vortex-hue <https://github.com/vortex-hue> Hey, its been pending to
> include demo-video. (as requested on TG)
> there are also now merge conflicts to resolve
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GoodDollar/GoodSDKs/pull/14?email_source=notifications&email_token=AXYCBIBFB2M6JR45QLI5E2D4TOSLTA5CNFSNUABFM5UWIORPF5TWS5BNNB2WEL2JONZXKZKDN5WW2ZLOOQXTIMJWGE4DQNRQGU42M4TFMFZW63VHNVSW45DJN5XKKZLWMVXHJNLQOJPWG33NNVSW45C7N5YGK3S7MNWGSY3L#issuecomment-4161886059>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXYCBIGAJWHYFWHBBLHH75L4TOSLTAVCNFSM6AAAAACD4EVGKWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHM2DCNRRHA4DMMBVHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

pheobeayo reviewed (changes requested):
**QA Report**
Env: Microsoft Edge & Chrome (desktop) / Ubuntu + Warpcast mobile app (Android) /
Wallet: MetaMask / Network: Celo mainnet
Branch: add-farcaster-support-for-the-identity-flow (commit 59484bc or later)

**Tests**

| Scenario | Expected | Actual | ✅/❌ |
|---|---|---|---|
| Desktop browser (Edge & Chrome) — outside Farcaster context | `isInFarcasterMiniApp()` returns `false`; normal (non-miniapp) verification flow loads and completes | Confirmed working in both browsers | ✅ |
| Farcaster mini app launch (Warpcast mobile, via ngrok tunnel + `fc:miniapp` embed) | App loads and identity verification UI becomes visible/interactive | Mini app launches with correct title ("GoodDollar Identity Demo") in header, but content area stays blank indefinitely (60+ sec, no change) | ❌ |
| Redirect handling (`rdu`/`cbu`) post-verification | N/A — blocked by above | Not reachable, could not test | ❌ (blocked) |

**Bugs**: 1 (Blocker — see separate comment below)
**Verdict**: Fail
**Evidence**:  https://www.loom.com/share/a9e494d15c494806ab28e7e67142e813, 
<img width="485" height="1080" alt="WhatsApp Image 2026-07-08 at 4 45 55 AM" src="https://github.com/user-attachments/assets/13b2284c-523a-4f8a-9e90-ad5748180047" />

L03TJ3 committed:
Merge branch 'finalize-farcaster-support' into add-farcaster-support-for-the-identity-flow

L03TJ3 reviewed (approved):
not approved but preparing for merge in local pull-request for copilot fixes
